Userland/Libraries/LibGUI/FileSystemModel.cpp

```diff
@@ -62,12 +62,13 @@ bool FileSystemModel::Node::fetch_data(String const& full_path, bool is_root)
 
     if (S_ISLNK(mode)) {
         auto sym_link_target_or_error = Core::File::read_link(full_path);
-        if (sym_link_target_or_error.is_error())
-            perror("readlink");
-
-        symlink_target = sym_link_target_or_error.release_value();
-        if (symlink_target.is_null())
+        if (sym_link_target_or_error.is_error()) {
```

Comment:
I am new here too but may I suggest using `TRY(Core::File::read_link(full_path));` This is more idiomatic and achieves the same objective
[TRY](https://github.com/SerenityOS/serenity/blob/HEAD/AK/Try.h#L12
)

Reply:
The function doesn't return an `ErrorOr<T>`, so `TRY()` won't work out-of-the-box. You would have to refactor the function a bit to make use of that.